src/App.js

```diff
@@ -1,11 +1,23 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import AddTooForm from './AddTodoForm';
 import TodoList from './TodoList';
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem("savedTodoList") || "[]"));
+  useEffect(function () {
+    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+  }, [todoList]);
+
+  return [todoList, setTodoList]
+}
+
 function App() {
 
-  
-  const [todoList, setTodoList] = useState([]);
+
+  const [todoList, setTodoList] = useSemiPersistentState(JSON.parse(localStorage.getItem("savedTodoList") || "[]"));
```

Comment:
What is the purpose of passing an argument to your `useSemiPersistentState` custom hook?